packages/esbuild-plugin-import-path-remapper/src/index.ts

```diff
@@ -0,0 +1,88 @@
+import { Plugin } from "esbuild";
+
+const resolvePath = (path: string, resolveDir: string): string | undefined => {
+  try {
+    return require.resolve(path, { paths: [resolveDir] });
+  } catch {
+    return undefined;
+  }
+};
+
+const findMainSourceFile = (
+  sourcePath: string,
+  resolveDir: string
+): string | undefined => {
+  const { main } = require(require.resolve(`${sourcePath}/package.json`, {
+    paths: [resolveDir],
+  }));
```

Comment:
Should we be using `resolvePath` here?

```suggestion
  const packageJsonPath = resolvePath(`${sourcePath}/package.json`, resolveDir);
  if (!packageJsonPath) {
    return undefined;
  }

  const { main } = require(packageJsonPath);
```

Reply:
`resolveDir` is esbuild's name for this (see where I get it from). Dir is more specific (path can be file or directory), and it aligns with the app, so doesn't it make sense to keep it?

Reply:
I meant `resolvePath` the function 😄 
https://github.com/microsoft/rnx-kit/blob/eb430eb9eddac23ceba428379b56e9a79a6f18bd/packages/esbuild-plugin-import-path-remapper/src/index.ts#L3

Reply:
Hehe, sorry. Yes, you're probably right ;)